### DISCUSSION TITLE
bug: terminal output at startup truncated

### DISCUSSION THREAD

**Discussion by ravibrock** [ghostty-org/ghostty#3938] [Feature Requests, Ideas] [answered]
It seems like Ghostty loads the shell session prior to resizing the window based on the `window-width` and `window-height` params. This means that if a command dependent on window size runs at shell startup (for example, i run `fastfetch` in my `.zshrc`) it ends up getting truncated unnecessarily. I have my terminal set to open to my display size by default, but I have the following output present in my terminal:
![Screenshot 2024-12-29 at 14 26 48](https://github.com/user-attachments/assets/e96096a5-8660-49d2-b7ee-0d758b49559a)

`ghostty --version` output:
```
Ghostty 1.0.0

Version
  - version: 1.0.0
  - channel: stable
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.none
  - font engine: font.main.Backend.coretext
  - renderer   : renderer.Metal
  - libxev     : main.Backend.kqueue
```

**Comment by mitchellh** [comment, marked as answer]
This was opened yesterday, feel free to add to that discussion. It's closed but I keep reading notifications. This is currently on purpose.
